### PR TITLE
Remove cluster-only stuff for new serverless version

### PR DESCRIPTION
**Pull request by calebwin**
This PR removes all but configuration stuff so that we can carefully implement (i.e., with proper testing): annotation API, banyan-polars, and session-related functions.

### CODE REVIEW

**Comment by cailinw** [comment]
@calebwin Can [res/pf_dispatch_table.toml](https://github.com/banyan-team/banyan-python/tree/caleb/remove-cluster-only-stuff/banyan/res)/pf_dispatch_table.toml be removed here?

File: banyan/banyan/imports.py [hunk 1]
```diff
@@ -5,13 +5,7 @@
 import time
 from typing import Any, Dict, Optional, Union, List
 
-# from _typeshed import NoneType
-
-# Third Party
 import boto3
 from botocore.config import Config
```
Comment (cailinw) [inline banyan/banyan/imports.py]: I think this can be removed? I don't see any place where this is used

**Review by cailinw — APPROVED** [review]
Great, thanks.

**Comment by calebwin** [comment]
> @calebwin Can [res/pf_dispatch_table.toml](https://github.com/banyan-team/banyan-python/tree/caleb/remove-cluster-only-stuff/banyan/res)/pf_dispatch_table.toml be removed here?

Yes! No PF dispatch table or PFs required - the only function to implement for each data type is a `convert_partition_type` to convert between different partition types (e.g., Blocked to Grouped or None to Blocked etc.)

**Pull request merged**